[Enhancement] [RHEL/7] New RHEL-7 remediation for the 'smartcard_auth' rule

iankko opened the pull request:
<br/>
</br/>

Testing report:
---------------
Verified manually on RHEL-7 system the proposed change works fine (puts formerly incompliant system wrt to this rule into compliance when performed).

Please review.

Thank you, Jan.
<br/>

redhatrises commented on RHEL/7/input/fixes/bash/smartcard_auth.sh:
```diff
@@ -0,0 +1,21 @@
+
+# Enable smartcard authentication (but allow also other ways
+# to login not to possibly cut off the system in question)
+authconfig --enablesmartcard
```
@iankko does `--update` or `--updateall` option need to be added to the authconfig command?

iankko commented:
@redhatrises 
> @iankko does --update or --updateall option need to be added to the authconfig command?

You are truly right. Thanks for catching this overlooking!
(on the system I was testing at has had smartcard auth enabled before => ```/etc/pam.d/system-auth``` etc. got modified before. Then only removed ```esc``` and ```pam_pkcs11``` packages to verify it works. But ```/etc/pam.d/system-auth``` was modified from the previous enablement yet. Your are correct ```authconfig``` needs either ```--update``` or ```--updateall``` to work properly. Went the ```--updateall``` way since IIUC two ```/etc/pam.d/``` files will be modified with mention about ```pam_pkcs11.so``` when smartcard auth is enabled).

Updated version above should fix this.

Thanks again for keeping eye on this!

iankko commented:
Just to mention it here. The remediation script as provided is causing openscap artificially to insert double quotes around the inserted value. This issue is not happening when running that Bash script directly, only when it's called via oscap as part of the remediation process.

I have tracked this issue upstream here:
&nbsp; &nbsp; [1] https://github.com/OpenSCAP/openscap/issues/174

Not sure we want to merge this PR (if so maybe only due to testing purposes for OpenSCAP). Looks the ```sed``` command at line 17 of this script will need to be rewritten into other form, not to artifically insert double quotes around the appended ```oscp_on``` value.

iankko commented:
Update on previous comment:
> Just to mention it here. The remediation script as provided is causing openscap artificially to insert double quotes around the inserted value. This issue is not happening when running that Bash script directly, only when it's called via oscap as part of the remediation process.

As commented in https://github.com/OpenSCAP/openscap/issues/174 the responsible component causing this issue doesn't seem to be openscap, but rather authconfig.

Details:
* issue doesn't appears when just sed part / sed script of the remediation is run within remediation (IOW ```authconfig --enablesmartcard --updateall``` command is not run),
* but it appears when both:
 * sed command and authconfig command are run together subsequently in the remediation script (it seems authconfig is trying to update the configuration of ```pam_pkcs11.conf``` configuration file yet [post sed processing] and within that effort inserting double quotes around the last field in the ```cert-policy``` key). => will file a new bug report against the authconfig component.

Dont' merge this PR yet though (I will update it yet so the double quote issue doesn't show when performing the remediation).

iankko commented:
The original ```smartcard_auth``` remediation script for RHEL-7 has been rewritten now the original:
<br/>
&nbsp; &nbsp; [1] https://github.com/OpenSCAP/openscap/issues/174
<br/>
not to exhibit:
* changed order in which the ```authconfig``` command and necessary ```sed``` scripts were executed,
* also simplified the ```sed``` scripts and inserted two tabs when modifying ```cert_policy``` key of ```/etc/pam_pkcs11/pam_pkcs11.conf``` file in order the final output form of this configuration file (after sed change) would match the general convention (two tabs used to indent second level of configuration).

Testing report:
-------------------
AFAICT from testing this updated version works fine for both scenarios:
* ```none``` value used for ```cert_policy``` key in ```/etc/pam_pkcs11/pam_pkcs11.conf``` file,
* ```[^oscp_on]``` value used for ```cert_policy``` key in ```/etc/pam_pkcs11/pam_pkcs11.conf``` file
and issue [1] doesn't exhibit with this implementation.

Feel free to re-review.

Thank you, Jan.

mpreisler commented:
It's a nasty workaround but it works. I will wait for Jenkins to finish but I don't expect issues.
